sdk/containerregistry/container-registry/src/blob/containerRegistryBlobClient.ts

```diff
@@ -0,0 +1,400 @@
+  public async uploadManifest(
+    manifest: (() => NodeJS.ReadableStream) | NodeJS.ReadableStream | OciManifest,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult> {
+    const { span, updatedOptions } = createSpan(
+      "ContainerRegistryBlobClient-uploadManifest",
+      options
+    );
+
+    try {
+      let manifestBody: Buffer | NodeJS.ReadableStream;
+
+      if (isReadableStream(manifest)) {
+        manifestBody = await readStreamToEnd(manifest);
+      } else if (typeof manifest === "function") {
+        manifestBody = await readStreamToEnd(manifest());
+      } else {
+        const serialized = serializer.serialize(Mappers.OCIManifest, manifest);
+        manifestBody = Buffer.from(JSON.stringify(serialized));
+      }
+
+      const tagOrDigest = options?.tag ?? (await calculateDigest(manifestBody));
+
+      const { dockerContentDigest } = await this.client.containerRegistry.createManifest(
+        this.repositoryName,
+        tagOrDigest,
+        manifestBody,
+        { contentType: KnownManifestMediaType.OciManifestMediaType, ...updatedOptions }
+      );
+
+      if (!dockerContentDigest) {
+        throw new Error("Digest not provided");
+      }
+
+      return { digest: dockerContentDigest };
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Downloads the manifest for an OCI artifact
+   *
+   * @param tagOrDigest - a tag or digest that identifies the artifact
+   * @returns - the downloaded manifest
+   */
+  public async downloadManifest(
+    tagOrDigest: string,
+    options?: DownloadManifestOptions
+  ): Promise<DownloadManifestResult> {
+    const { span, updatedOptions } = createSpan(
+      "ContainerRegistryBlobClient-downloadManifest",
+      options
+    );
+
+    try {
+      const rawResponse = await new Promise<FullOperationResponse>(async (resolve) => {
+        this.client.containerRegistry.getManifest(this.repositoryName, tagOrDigest, {
+          accept: KnownManifestMediaType.OciManifestMediaType,
+          ...updatedOptions,
+          onResponse: (rawResponseProvided, flatResponse, error) => {
+            updatedOptions?.onResponse?.(rawResponseProvided, flatResponse, error);
+            resolve(rawResponseProvided);
+          },
+        });
+      });
+
+      const digest = rawResponse.headers.get("Docker-Content-Digest");
+      if (!digest) {
+        throw new Error("Expected Docker-Content-Digest header in response");
+      }
+
+      const body = rawResponse.bodyAsText;
+      if (!body) {
+        throw new Error("downloadManifest did not return a text body");
+      }
+
+      const bodyData = Buffer.from(body);
+      const expectedDigest = await calculateDigest(bodyData);
+
+      if (digest !== expectedDigest) {
+        throw new Error(`Docker-Content-Digest header does not match calculated digest.
+        Expected: ${expectedDigest}
```

Comment:
do the actual/expected values actually tell the dev anything useful?

Reply:
It was useful to me during testing but during normal operation probably not. I can remove it.